libraries/AP_HAL_ChibiOS/hwdef/JHEMCU-GSF405A/hwdef.dat

```diff
@@ -0,0 +1,165 @@
+# hw definition file for processing by chibios_pins.py
+# JHEMCU GSF405A by V-22
+# With F405 MCU, MPU6000 IMU and MAX7456 series OSD
+# Based on Mamba F405 from jeanphilippehell
+@ thanks to betaflight for pin information
+
+MCU STM32F4xx STM32F405xx
+
+# board ID for firmware load
+APJ_BOARD_ID 1060
+
+# crystal frequency
+OSCILLATOR_HZ 8000000
+
+# board voltage
+STM32_VDD 330U
+
+# need to use timer 4 so timer 5 is available for bidir DShot
+define STM32_ST_USE_TIMER 4
+define CH_CFG_ST_RESOLUTION 16
+
+# order of I2C buses
+I2C_ORDER I2C1
+
+# order of UARTs (and USB)
+# this order follows the labels on the board
+SERIAL_ORDER OTG1 USART1 USART2 USART3 UART4 EMPTY USART6
+
+# The pins that USB is connected on
+PA11 OTG_FS_DM OTG1
+PA12 OTG_FS_DP OTG1
+
+# USB detection
+PA8 VBUS INPUT OPENDRAIN
+
+# USART1 (DMA) - wired to on-board ELRS receiver - need DMA for CRSF protocol
+PB6 USART1_TX USART1
+PA10 USART1_RX USART1
+
+# default Serial1 to ELRS RX
+define HAL_SERIAL1_PROTOCOL 23
+define HAL_RSSI_TYPE 3 
+
+# Alt config to allow PPM on RX1
+#PA10 TIM1_CH3 TIM1 RCININT PULLDOWN ALT(1)
```

Comment:
You should just remove this

Reply:
Is anything required to enable SBUS input on the SBUS pin? It looks like it is connected to RX2 (USART2_RX) on pin D6, which has no timer resources available so I don't think we can use it for PPM.

Reply:
This is an F4, so it should have an inversion pin (look in BF), you define it using something like:
```
# SBUS inversion control pin, active high
PB15 USART3_RXINV OUTPUT LOW GPIO(78) POL(1)
```

Reply:
It doesn't have an inversion pin. Since it has separate pads for SBUS and RX2 I assume the SBUS pad is just connected through a hardware inverter.

Reply:
Ok in which case it looks like its connected to UART2 in which case you really need DMA on this UART as there is no other way to get SBUS